src/mono/browser/runtime/hybrid-globalization/locales-common.ts

```diff
@@ -0,0 +1,78 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { VoidPtrNull } from "../types/internal";
+import { Int32Ptr, VoidPtr } from "../types/emscripten";
+import { OUTER_SEPARATOR, normalizeLocale } from "./helpers";
```

Comment:
I believe this is also part of dotnet.runtime.js module and so it should be one directory up. The tricky part is the re-use of `./helpers` in both modules. How big code we are pulling in twice ?

Reply:
`OUTER_SEPARATOR` is just a char. The method is a few lines:
```
export function normalizeLocale (locale: string | null) {
    if (!locale)
        return undefined;
    try {
        locale = locale.toLocaleLowerCase();
        if (locale.includes("zh")) {
            // browser does not recognize "zh-chs" and "zh-cht" as equivalents of "zh-HANS" "zh-HANT", we are helping, otherwise
            // it would throw on getCanonicalLocales with "RangeError: Incorrect locale information provided"
            locale = locale.replace("chs", "HANS").replace("cht", "HANT");
        }
        const canonicalLocales = (Intl as any).getCanonicalLocales(locale.replace("_", "-"));
        return canonicalLocales.length > 0 ? canonicalLocales[0] : undefined;
    } catch {
        return undefined;
    }
}
```